### PR TITLE
Support preset modes and percentage for mqtt fan

### PR DESCRIPTION
**Pull request by jbouwh**
## Proposed change

The fan entity model has changed. The code and documentation has updated to support fan speed percentage and preset mode functions. Using legacy speeds is deprecated.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [x] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase:  (https://github.com/home-assistant/core/pull/47944)
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/fan.mqtt.markdown [hunk 1]
```diff
@@ -129,17 +129,17 @@ payload_available:
   type: string
   default: online
 payload_high_speed:
-  description: The payload that represents the fan's high speed.
+  description: DEPRECATED - The payload that represents the fan's high speed.
```
Comment (emontnemery) [inline source/_integrations/fan.mqtt.markdown]: Remove all the deprecated configuration options.

File: source/_docs/mqtt/discovery.markdown [hunk 2]
```diff
@@ -192,6 +195,10 @@ Supported abbreviations:
     'pow_cmd_t':           'power_command_topic',
     'pow_stat_t':          'power_state_topic',
     'pow_stat_tpl':        'power_state_template',
+    'pr_mode_cmd_t':       'preset_mode_command_topic',
+    'pr_mode_stat_t':      'preset_mode_state_topic',
+    'pr_mode_value_tpl':   'preset_mode_value_template',
```
Comment (jbouwh) [inline source/_docs/mqtt/discovery.markdown]: 'pr_mode_value_tpl' should be 'pr_mode_val_tpl'

This was already checked in abbreviations.py

**Review by jbouwh — COMMENTED** [review]
One abbreviation 'pr_mode_value_tpl' should be corrected,

File: source/_docs/mqtt/discovery.markdown (hunk 2 above)
Comment (jbouwh) [reply source/_docs/mqtt/discovery.markdown]: Commited the fix

**Review by emontnemery — APPROVED** [review]
Looks good!

**Comment by emontnemery** [comment]
There was a mistake here, this should have targetted `next` branch, not `current` (core PR is not in 2021.3, but will be in 2021.4).

**Comment by frenck** [comment]
Will adjust this manually

**Comment by frenck** [comment]
Done, it has been reverted on `current` and is now re-applied on `next`.

**Pull request merged**